crates/pg_workspace_new/src/workspace/server/change.rs

```diff
@@ -122,13 +122,22 @@ impl Document {
         // special case: if no statement is affected, the affected range is between the prev and
         // the next statement
         if affected.is_empty() {
-            let start = self
+            // since we do not now whether the change should be part of the previous statement, we
+            // will take the range form the start of the previous statement to the start of the
```

Comment:
We also need to take the next statement into consideration and then compare the resulting range to check weiter it's the prev or next stmt